core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -30,11 +48,7 @@ private[chisel3] object PropertyType {}
   * describe a set of non-hardware types, so they have no width, cannot be used
   * in aggregate Data types, and cannot be connected to Data types.
   */
-class Property[T: PropertyType: TypeTag] extends BaseType {
-
-  /** A TypeTag for the type T of this Property.
-    */
-  private[chisel3] val tpe = typeOf[T]
+class Property[T](implicit propertyType: PropertyType[T]) extends BaseType {
```

Comment:
The sugar way of doing this is:

```suggestion
class Property[T : PropertyType] extends BaseType {
```